compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -2991,6 +2996,33 @@ object TypeComparer {
   end ApproxState
   type ApproxState = ApproxState.Repr
 
+  /** Result of `isCovered` check. */
+  object CoveredStatus:
+    opaque type Repr = Int
+
+    private inline val IsCovered = 2
+    private inline val NotHasOr = 1
+
+    /** The type is not covered. */
+    val Uncovered: Repr = 1
```

Comment:
Style: In this case I'd write the three vals on subsequent lines with // comments to the right. It's more compact taht way and just as legible.